sdk/textanalytics/ai-text-analytics/src/util.ts

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { FullOperationResponse, OperationOptions } from "@azure/core-client";
+import { LanguageDetectionInput, TextDocumentInput } from "./generated";
+import { LroResponse } from "@azure/core-lro";
+import { TextAnalysisOperationOptions } from "./models";
+import { logger } from "./logger";
+
+interface IdObject {
+  id: string;
+}
+
+/**
+ * Given a sorted array of input objects (with a unique ID) and an unsorted array of results,
+ * return a sorted array of results.
+ *
+ * @internal
+ * @param sortedArray - An array of entries sorted by `id`
+ * @param unsortedArray - An array of entries that contain `id` but are not sorted
+ */
+export function sortResponseIdObjects<T extends IdObject, U extends IdObject>(
+  sortedArray: T[],
```

Comment:
Is the instantiation of `T` actually important here? In other words, could this be:

```suggestion
  sortedArray: IdObject[],
```

Reply:
No, it is not. Done.